### PR TITLE
Feature/tutorial matt (Add ratio check filter to UFO)

### PR DESCRIPTION
**Pull request by MatthewMorris-NOAA**
Added a new ratio check filter to UFO, as well as a new ctest (qc_ratiocheck) to test the new filter.

### CODE REVIEW

File: src/ufo/filters/RatioCheck.cc [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  std::vector<float> ref, val;
+  data_.get(ref_, ref);
+  data_.get(val_, val);
+  ASSERT(ref.size() == val.size());
+
+// Loop over all obs
+  for (size_t jobs = 0; jobs < nlocs; ++jobs) {
+    if (apply[jobs]) {
+      // check to see if one of the reference or value is missing
+      if (val[jobs] == missing || ref[jobs] == missing || ref[jobs] == 0) {
```
Comment (emilyhcliu) [inline src/ufo/filters/RatioCheck.cc:68]: @MatthewMorris-NOAA The vector ref has a float type, so it is better to change 0 to 0.0.

File: src/ufo/filters/QCflags.h [hunk 2]
```diff
@@ -27,6 +27,7 @@ namespace QCflags {
   constexpr int buddy   = 13;  // observation rejected by the buddy check
   constexpr int derivative = 14;  // observation removed due to metadata derivative value
   constexpr int profile = 15;  // observation rejected by at least one profile QC check
+  constexpr int ratioref = 16;   // ratio of two values outside of range
   constexpr int onedvar = 16;  // observation failed to converge in 1dvar check
```
Comment (emilyhcliu) [inline src/ufo/filters/QCflags.h:31]: @MatthewMorris-NOAA There are two 16s here.  It is better to make our new filter as 17.

File: src/ufo/filters/RatioCheck.cc (hunk 1 above)
Comment (MatthewMorris-NOAA) [reply src/ufo/filters/RatioCheck.cc]: Updated.

File: src/ufo/filters/QCflags.h (hunk 2 above)
Comment (MatthewMorris-NOAA) [reply src/ufo/filters/QCflags.h:30]: Updated.

**Review by emilyhcliu — APPROVED** [review]

**Pull request closed without merging**